### DISCUSSION TITLE
Is it possible to export a note to MD without links to pdf automatically

### DISCUSSION THREAD

**Discussion by Panipenem** [windingwind/zotero-better-notes#1160] [Q&A]
![2024-10-14 175217](https://github.com/user-attachments/assets/cd37a9c6-558a-4aad-bb76-5ab7d7138e87)

When exporting notes to MD, the citations retain links to their original positions in the PDF, as shown in the second paragraph of the image above. I found that I can manually unlink each citation in Zotero Better Notes before exporting to get the format of the first paragraph. I understand using Zotero's native export feature can get the desired result, but it doesn't support auto-sync and pictures exports.

Is there an automated method to remove these links?

Thx

**Comment by windingwind** [comment]
It's not possible to do syncing without links. The information is permanently lost when you unlink the citations and links.

**Reply by Panipenem** [reply]
Thank you for your timely reply. The link in the citation part is enough for me to find the original literature in the Zotero library. I am unsure what information of the link on the citation text would be lost should I unlink the pdf in the better notes. Could you please elaborate on this, like info for the specific location in the pdf file? Thx in advance.

**Reply by windingwind** [reply]
It would be weird if the exported citations in markdown are unlinked and the original ones in the note are not. It's not right to unlink the citations in the note when syncing back from the markdown, but it's also not right to keep them linked, as they will be converted to citations with links the next time you sync from note to markdown.

If you really want to do that, you can edit the `ExportMDFileContent` template and remove all the link's `href`.

**Reply by Dani-v-tue** [reply]
In case anyone else runs into this and is looking for a snippet, I used the code below in `ExportMDFileContent` to ensure that all exported links indeed became markdown links and to remove other unrelated HTML tags that somehow appeared in my markdown.  
I'm sure it is adaptable to other uses.  

```
${{
  // Use a regex to match HTML <a> tags and convert them to Markdown links.
  // This regex matches both single and double quotes, and handles additional attributes gracefully.
  const htmlLinkRegex = /<a\s+href=["']([^"']+)["'][^>]*>(.*?)<\/a>/gi;
  mdContent = mdContent.replace(htmlLinkRegex, "[$2]($1)");

  // Remove any remaining HTML tags (e.g., <span>, </span>, etc.)
  // This regex matches any opening or closing tag and removes it.
  mdContent = mdContent.replace(/<\/?[^>]+>/gi, "");

  return mdContent; //The original just had this line
}}$
```

**Reply by charle36** [reply]
Thanks a lot Dani!

I was able to use your template to format quotes as list items, links as buttons next to the quote, and comments as indented list items. Combined with the template removing empty spaces, here's the result: 

```
${{
  // Use a regex to match HTML <a> tags and convert them to Markdown links.
  // This regex matches both single and double quotes, and handles additional attributes gracefully.
  const htmlLinkRegex = /<a\s+href=["']([^"']+)["'][^>]*>(.*?)<\/a>/gi;
  mdContent = mdContent.replace(htmlLinkRegex, "- $2 [🅉]($1) ");

  // Remove any remaining HTML tags (e.g., <span>, </span>, etc.)
  // This regex matches any opening or closing tag and removes it.
  mdContent = mdContent.replace(/<\/?[^>]+>/gi, "");

// This regex matches citation info and removes it.
  mdContent = mdContent.replace(/ \(\- .*p\. \d* \[..\]\(zotero:\/\/select\/library\/items\/\w*\) \)/gi, "");

// This regex matches format comments as indented list items.
  mdContent = mdContent.replace(/\)  /gi, ") \n	- ");

// This part removes empty lines.
	let lines = mdContent.split("\n");
  	function isListLine(line) {
  		return /^(?:\*|-|\+|[0-9]+\.)(?= )/.test(line.trim());
  	}
  	let filteredLines = [];
  	for (let i = 0; i < lines.length; i++) {
  		let currentLine = lines[i];
  		if (i > 0 && i < lines.length - 1) {
  			// Not first or last line. Check empty lines between item list lines
  			let previousLine = lines[i-1];
  			let nextLine = lines[i+1];
  			if (currentLine.trim().length === 0) {
  					// Skip empty line
  					continue;
  			}
  		}
  		filteredLines.push(currentLine);
  	}
  	return filteredLines.join("\n").replace(/\\-/g, "-").replace(/\\:/g, ": ").replace(/^(\s*)\* /gm, "$1-");
  }}$
```

It's kind of messy, but I thought I would share anyway since it might help someone.
